src/compiler/transformers/destructuring.ts

```diff
@@ -521,8 +521,10 @@ namespace ts {
                 for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                     t[p] = s[p];
                 if (s != null && typeof Object.getOwnPropertySymbols === "function")
-                    for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) if (e.indexOf(p[i]) < 0)
-                        t[p[i]] = s[p[i]];
+                    for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
+                        if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable(p[i]))

```

Comment:
This is incorrect as you are passing the wrong receiver. Its highly unlikely `p[i]` is enumerable on `Object.prototype`. You probably mean this:
```suggestion
                        if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
```

Reply:
Ah, makes sense. In that case, does `s.propertyIsEnumerable(p[i])` work as well?

Reply:
On second thought, I suppose `s` could have its own implementation of propertyIsEnumerable, so no, that probably does not work.